analyzers/tests/SonarAnalyzer.UnitTest/Facade/SyntaxFacadeTest.cs

```diff
@@ -93,5 +93,19 @@ public class SyntaxFacadeTest
         [TestMethod]
         public void NodeStringTextValue_UnexpectedType_VB() =>
             vb.NodeStringTextValue(VB.SyntaxFactory.ThrowStatement()).Should().BeEmpty();
+
+        [TestMethod]
+        public void RemoveConditionalAccess_Null_CS() =>
+            cs.RemoveConditionalAccess(null).Should().BeNull();
+
+        [DataTestMethod]
+        [DataRow("M()", "M()")]
+        [DataRow("this.M()", "this.M()")]
+        [DataRow("A.B.C.M()", "A.B.C.M()")]
+        [DataRow("A.B?.C.M()", ".C.M()")]
+        [DataRow("A.B?.C?.M()", ".M()")]
+        [DataRow("A.B?.C?.D", ".D")]
+        public void RemoveConditionalAccess_SimpleInvocation_CS(string invocation, string expected) =>
```

Comment:
[education] @martin-strecker-sonarsource what's the usefulness of this method, what's a scenario when this is useful?

Reply:
ConditionalAccess messes up the syntax tree because of left/right associativity of the `.` operator vs. the `?.` operator (I don't remember the details, but there is a 200+ thread on csharplang discussing the pros and cons). Do get an idea about the complexity involved you may want to have a look at:

https://github.com/dotnet/roslyn/blob/2aed89bfee5d5318cb62108be76365befb656c38/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs#L295-L371

This helper method simplifies any invocation to get just the "simple" part without the conditionals messing up the syntax tree. It is easier to get the Method symbol if the tree is cleaned up.

What we should actually do is to use GetParentConditionalAccessExpression of Roslyn, but I did not have looked whether this (and some other related helpers) are already present in our code base.

Reply:
thanks!